File: 5 - JAVA e PHP (POO)/PHP/Aula14_Exercicio_Parte_01/Pessoa.php
<?php

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of Pessoa
 *
 * @author ronan
 */
abstract class Pessoa {
    //Atributos
    protected $nome;
    protected $idade;
    protected $sexo;
    protected $experiencia;
    
    //Construtor
    public function __construct($nome,$idade,$sexo){
        $this->setNome($nome);
        $this->setIdade($idade);
        $this->setSexo($sexo);
        $this->setExperiencia(0);
    }
    
    //Getters and Setters
    function getNome() {
        return $this->nome;
    }

    function getIdade() {
        return $this->idade;
    }

    function getSexo() {
        return $this->sexo;
    }

    function getExperiencia() {
        return $this->experiencia;
    }

    function setNome($nome) {
        $this->nome = $nome;
    }

    function setIdade($idade) {
        $this->idade = $idade;
    }

    function setSexo($sexo) {
        $this->sexo = $sexo;
    }

    function setExperiencia($experiencia) {
        $this->experiencia = $experiencia;
    }

    //Método Abstrato
    public abstract function ganharExperiencia();

}

File: 5 - JAVA e PHP (POO)/JAVA/Aula15_Exercicio_Parte_02_Youtube/Youtube_Parte_02/src/youtube_parte_02/Pessoa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package youtube_parte_02;

/**
 *
 * @author ronan
 */
public abstract class Pessoa {
    //Atributos
    protected String nome;
    protected int idade;
    protected char sexo;
    protected int experiencia;
    
    //Construtor
    public Pessoa(String nome,int idade,char sexo){
        this.setNome(nome);
        this.setIdade(idade);
        this.setSexo(sexo);
        this.setExperiencia(0);
    }
    
    //Getters and Setters

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getIdade() {
        return idade;
    }

    public void setIdade(int idade) {
        this.idade = idade;
    }

    public char getSexo() {
        return sexo;
    }

    public void setSexo(char sexo) {
        this.sexo = sexo;
    }

    public int getExperiencia() {
        return experiencia;
    }

    public void setExperiencia(int experiencia) {
        this.experiencia = experiencia;
    }
    
    //Método Abstrato
    protected abstract void ganharExperiencia();
    protected abstract void apresentarGafanhoto();
}
